File: backend/Models/Ad.cs
// Models/Ad.cs
namespace Avukado.Api.Models;

public class Ad
{
    public int Id { get; set; }
    public string Title { get; set; } = "";
    public string Description { get; set; } = "";
    public string City { get; set; } = "";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public List<Proposal> Proposals { get; set; } = new();
}

public class Proposal
{
    public int Id { get; set; }
    public int AdId { get; set; }
    public Ad? Ad { get; set; }
    public int LawyerId { get; set; }
    public Lawyer? Lawyer { get; set; }
    public decimal Price { get; set; }
    public string Status { get; set; } = "pending"; // pending|accepted|rejected
}

File: backend/src/models/Ad.js
import mongoose from "mongoose";

const adSchema = new mongoose.Schema(
    {
        title: {
            type: String,
            required: true,
        },
        description: {
            type: String,
            required: true,
        },
        category: {
            type: String,
            required: true,
            enum: [
                "ceza",
                "ticaret",
                "miras",
                "is-hukuku",
                "icra-iflas",
                "gayrimenkul",
                "bilisim",
                "tuketici",
                "vergi",
                "aile",
            ],
        },
        city: {
            type: String,
            required: true,
        },
        client: {
            type: mongoose.Schema.Types.ObjectId,
            ref: "User",
            required: true,
        },
        status: {
            type: String,
            enum: ["open", "closed", "in-progress"],
            default: "open",
        },
        documents: [
            {
                name: String,
                url: String,
                uploadedAt: { type: Date, default: Date.now },
            },
        ],
        budget: {
            type: Number,
        },
        proposals: [
            {
                type: mongoose.Schema.Types.ObjectId,
                ref: "Proposal",
            },
        ],
    },
    {
        timestamps: true,
    }
);

export default mongoose.model("Ad", adSchema);

